apps/studio-next/src/app/api/v1/generate/route.tsx

```diff
@@ -0,0 +1,6 @@
+import { NextResponse } from 'next/server';
```

Comment:
Not sure if we can use [Server Actions](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations) instead 🤔 ?

Reply:
@jerensl

I searched the Internet back and forth and couldn't find answers to several simple questions.
You [seem to be](https://github.com/asyncapi/studio/issues/1118) very knowledgeable in Next.js Server's matters, so I would like to consult with you on this topic.

Server Actions are meant for performing data fetching and mutations on the Next.js Server.
To my understanding, this implies that Server ALREADY exists as an independent entity, in order for a Server Action to have an environment to be executed in.
In this particular case, right now, full Server consists of one API route.
Thus, if this one and only API route is replaced by a Server Action, the Server will simply vanish, and Server Action will have no place to be executed in.
Besides, this API route is responsible for Server Response, which requires exactly the `next/server` package, and not for data fetching from a third-party resource or a mutation.

Taking this into account, does Server Action even make sense in this case, or should the structure with one API route remain (for now, the final architecture of the `Studio`'s Server-Side is not defined yet, and this route is added only as a Proof Of Concept and a stub in place where the future Server will be?)

Please correct me if I'm getting something wrong in the new for me concept of Server Actions.

Reply:
@aeworxet 

> I searched the Internet back and forth and couldn't find answers to several simple questions. You [seem to be](https://github.com/asyncapi/studio/issues/1118) very knowledgeable in Next.js Server's matters, so I would like to consult with you on this topic.

I still new to this Reace Server Components (RSC), so I will answers as much as I know
 
> Server Actions are meant for performing data fetching and mutations on the Next.js Server. To my understanding, this implies that Server ALREADY exists as an independent entity, in order for a Server Action to have an environment to be executed in. In this particular case, right now, full Server consists of one API route. Thus, if this one and only API route is replaced by a Server Action, the Server will simply vanish, and Server Action will have no place to be executed in. Besides, this API route is responsible for Server Response, which requires exactly the `next/server` package, and not for data fetching from a third-party resource or a mutation.
> 
> Taking this into account, does Server Action even make sense in this case, or should the structure with one API route remain (for now, the final architecture of the `Studio`'s Server-Side is not defined yet, and this route is added only as a Proof Of Concept and a stub in place where the future Server will be?)

The thing about [Api Route](https://nextjs.org/docs/pages/building-your-application/routing/api-routes) which their rebranding as [Route Handler](https://nextjs.org/docs/app/building-your-application/routing/route-handlers) is not supposed to be replaced by Server Actions but different type options in the Architecture which would be considered during the design

https://www.youtube.com/watch?v=PtX5PF17tlQ

> Please correct me if I'm getting something wrong in the new for me concept of Server Actions.

Server Actions are supposed to work on top of the React Server Component(RSC) but the majority of the applications inside [StudioWrapper](https://github.com/asyncapi/studio/blob/master/apps/studio-next/src/components/StudioWrapper.tsx) do not leverage RSC even for NextJS features like SSR/SSG do not work there

Static Side Generator -> render JSX to HTML during built time
Server Side Rendering -> render JSX to HTML at server when requested by a Client (give what u need only)

This one of the good resources Dan Abramov explains what are the responsibilities of SSR(What NextJS did) and RSC, and how they work together interchangeably
https://github.com/reactwg/server-components/discussions/5

Reply:
@KhudaDad414 
Would it be safe to assume that Server Actions should not be used currently?

Reply:
@aeworxet you are right, we first need to decide how we are going to architecture the component for the `app` directory.